lucet-runtime/lucet-runtime-internals/src/module/dl.rs

```diff
@@ -71,31 +69,43 @@ impl DlModule {
             std::ptr::null()
         };
 
-        let trap_manifest = unsafe {
-            if let Ok(len_ptr) = lib.get::<*const u32>(b"lucet_trap_manifest_len") {
-                let len = len_ptr.as_ref().ok_or(lucet_incorrect_module!(
-                    "`lucet_trap_manifest_len` is defined but null"
-                ))?;
-                let records = lib
-                    .get::<*const TrapManifestRecord>(b"lucet_trap_manifest")
-                    .map_err(|e| {
-                        lucet_incorrect_module!("error loading symbol `lucet_trap_manifest`: {}", e)
-                    })?
-                    .as_ref()
-                    .ok_or(lucet_incorrect_module!(
-                        "`lucet_trap_manifest` is defined but null"
+        let function_manifest = unsafe {
+            let manifest_len_ptr = lib.get::<*const u32>(b"lucet_function_manifest_len");
+            let manifest_ptr = lib.get::<*const FunctionSpec>(b"lucet_function_manifest");
+
+            match (manifest_ptr, manifest_len_ptr) {
+                (Ok(ptr), Ok(len_ptr)) => {
+                    let manifest_len = len_ptr.as_ref().ok_or(lucet_incorrect_module!(
+                        "`lucet_function_manifest_len` is defined but null"
                     ))?;
-                from_raw_parts(records, *len as usize)
-            } else {
-                &[]
+                    let manifest = ptr.as_ref().ok_or(lucet_incorrect_module!(
+                        "`lucet_function_manifest` is defined but null"
+                    ))?;
+
+                    from_raw_parts(manifest, *manifest_len as usize)
+                }
+                // TODO: Can a module be data-only?
```

Comment:
technically, it is valid for a webassembly module to define no functions

Reply:
cool. adjusted this match arm in light of that